### PR TITLE
[ADD] product_supplierinfo_for_customer_elaboration: New module to set default elaborations

### PR DESCRIPTION
**Pull request by CarlosRoca13**
cc @Tecnativa TT37184

please @carlosdauden @sergio-teruel review this

### CODE REVIEW

File: product_supplierinfo_for_customer_elaboration/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Copyright 2022 Tecnativa - Carlos Roca
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+{
+    "name": "Product Supplierinfo for Customers Elaboration",
+    "summary": "Allows to define default elaborations and elaboration notes on a ",
```
Comment (sergio-teruel) [inline product_supplierinfo_for_customer_elaboration/__manifest__.py]: "on a...."

**Review by sergio-teruel — CHANGES_REQUESTED** [review]

File: product_supplierinfo_for_customer_elaboration/models/sale_order_line.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        res = super().product_id_change()
+        customerinfo = self.product_id.customer_ids.filtered(
+            lambda pc: pc.elaboration_id and pc.name == self.order_id.partner_id
+        )[:1]
+        if not customerinfo:
+            customerinfo = customerinfo.filtered(
+                lambda pc: pc.elaboration_id
+                and pc.name == self.order_id.partner_id.commercial_partner_id
+            )[:1]
+        self.elaboration_id = customerinfo.elaboration_id
```
Comment (carlosdauden) [inline product_supplierinfo_for_customer_elaboration/models/sale_order_line.py]:
```suggestion
        if customerinfo:
            self.elaboration_id = customerinfo.elaboration_id
```

**Comment by CarlosRoca13** [comment]
@sergio-teruel @carlosdauden changes done

File: product_supplierinfo_for_customer_elaboration/models/sale_order_line.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        customerinfo = self.product_id.customer_ids.filtered(
+            lambda pc: pc.elaboration_id and pc.name == self.order_id.partner_id
+        )[:1]
+        if not customerinfo:
+            customerinfo = customerinfo.filtered(
+                lambda pc: pc.elaboration_id
+                and pc.name == self.order_id.partner_id.commercial_partner_id
+            )[:1]
+        if customerinfo:
+            self.elaboration_id = customerinfo.elaboration_id
```
Comment (cesar-tecnativa) [inline product_supplierinfo_for_customer_elaboration/models/sale_order_line.py:22]: return res?

**Review by cesar-tecnativa — CHANGES_REQUESTED** [review]

File: product_supplierinfo_for_customer_elaboration/models/sale_order_line.py (hunk 3 above)
Comment (CarlosRoca13) [reply product_supplierinfo_for_customer_elaboration/models/sale_order_line.py:22]: https://github.com/OCA/sale-workflow/pull/2087/commits/9d5623305b353d9019909b1fba116544d2bde867#diff-8c2de5eca200a60ed3c716e555be7524c0ece8b07c527040c321556af36f0ad3R25

Comment (cesar-tecnativa) [reply product_supplierinfo_for_customer_elaboration/models/sale_order_line.py:22]: :+1:

**Review by cesar-tecnativa — APPROVED** [review]

File: product_supplierinfo_for_customer_elaboration/models/sale_order_line.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    @api.onchange("product_id")
+    def product_id_change(self):
+        res = super().product_id_change()
+        customerinfo = self.product_id.customer_ids.filtered(
+            lambda pc: pc.elaboration_id and pc.name == self.order_id.partner_id
+        )[:1]
+        if not customerinfo:
```
Comment (pedrobaeza) [inline product_supplierinfo_for_customer_elaboration/models/sale_order_line.py:16]: This doesn't make sense, as if no `customerinfo`, filter anything will be again no `customerinfo`

Comment (pedrobaeza) [reply product_supplierinfo_for_customer_elaboration/models/sale_order_line.py:16]: Can you please check this?

**Comment by CarlosRoca13** [comment]
Changed @pedrobaeza

**Review by sergio-teruel — APPROVED** [review]

**Review by pedrobaeza — COMMENTED** [review]
No tests?

**Comment by CarlosRoca13** [comment]
Added @pedrobaeza

**Pull request merged**